webapp/channels/src/components/dynamic_virtualized_list/item_row_shared.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import debounce from 'lodash/debounce';
+import type {ReactNode} from 'react';
+import React, {memo, useEffect, useRef} from 'react';
+
+import {ListItemSizeObserver} from './item_row_size_observer';
+
+const RESIZE_DEBOUNCE_TIME = 200; // in ms
+
+const listItemSizeObserver = new ListItemSizeObserver();
```

Comment:
I am not sure if this an anti-pattern or not, but we usually try to avoid global variables on the server side. We could have a singleton class with a static method to return the observer:

```js
class ListItemObserverService {
  static instance = null;  // Static property
  
  static getInstance() {   // Static method
    if (!ListItemObserverService.instance) {
      ListItemObserverService.instance = new ListItemObserverService();
    }
    return ListItemObserverService.instance;
  }
  
  // Instance methods and properties...
}
```

0/5